pallets/parachain-staking/src/lib.rs

```diff
@@ -1581,12 +1434,11 @@ pub mod pallet {
 
 			// update state and potentially prepare kicking a delegator with less staked
 			// amount (includes setting rewards for kicked delegator)
-			let (state, maybe_kicked_delegator) = if num_delegations_pre_insertion == T::MaxDelegatorsPerCollator::get()
-			{
+			let state = if num_delegations_pre_insertion == T::MaxDelegatorsPerCollator::get() {
 				Self::do_update_delegator(delegation, state)?
 			} else {
 				state.total = state.total.saturating_add(amount);
-				(state, None)
```

Comment:
We don't need to care about the kicked delegator anymore because they cannot have other delegations anymore.